torch/onnx/_internal/exporter/_compat.py

```diff
@@ -153,12 +157,26 @@ def export_compat(
     else:
         args, kwargs = _get_torch_export_args(args, kwargs)
         if dynamic_shapes is None and dynamic_axes is not None:
-            dynamic_shapes = _from_dynamic_axes_to_dynamic_shapes(
-                model,
-                dynamic_axes=dynamic_axes,
-                input_names=input_names,
-                output_names=set(output_names or ()),
+            warnings.warn(
+                "# ⚠️ **dynamic_axes is not recommended to dynamo=True, "
+                "and might lead to torch._dynamo.exc.UserError: Constraints violated. "
+                "Please use dynamic_shapes for better user experience.** ⚠️",
+                UserWarning,
             )
+            try:
+                dynamic_shapes = _from_dynamic_axes_to_dynamic_shapes(
+                    model,
+                    args,
+                    kwargs,
+                    dynamic_axes=dynamic_axes,
+                    input_names=input_names,
+                    output_names=set(output_names or ()),
+                )
+            except Exception as e:
+                raise RuntimeError(
+                    "# ❌ **Failed to convert dynamic_axes to dynamic_shapes. "
+                    "Please provide dynamic_shapes directly."
```

Comment:
Since this is already an error, we can keep it consistent with other python errors. We can save the emojis for status information in prints.

```suggestion
                    "# Failed to convert 'dynamic_axes' to 'dynamic_shapes'. "
                    "Please provide 'dynamic_shapes' directly. "
                    "Refer to the documentation for 'torch.export.export' for more information on dynamic shapes."
```